portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,10 +32,18 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
  
-  private ArrayList<String> comments = new ArrayList<String>();
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    
+    ArrayList<String> comments = new ArrayList<String>();
```

Comment:
This is mostly a nit, but I'd try to avoid using ArrayList (or other specific implementations) as types. Nothing in your code is dependent on it being an array.

Two main reasons for this:
A: it's easier to read, List is more intuitive to read than ArrayList, and might lead reviewers to look for why it's being used specifically.
B: it's more maintainable, if for some reason you had to switch to a different type of list, you wouldn't need to update the code.

This advice is particularly true for method stubs, I also try to use the bare minimum type (ex: give what you're doing, it would actually be Iterable<String>, but I would typically use Collection<> over that, although I think List<> is perfectly fine here)

A concrete example of when I'd use a specific sub-type is when using ImmutableList. In this case it's enforcing that the the data will not change after the object is constructed. As a method parameter, this would would prevent mutable lists from being passed in, which when dealing with concurrency may actually be important for simple actions like looping over the list

Reply:
Just a minor follow up for clarity: I mentioned using iterable and collection, in this case the order is relevant, so I would definitely stick with List<String> as the others are not necessarily ordered